### PR TITLE
Add batch_generate method to ModelAdapter for concurrent prompt proce…

### PR DESCRIPTION
**Pull request by yanurag-dev**
…ssing

Implemented the batch_generate method in the ModelAdapter class to handle generating responses for multiple prompts concurrently, with rate limit protection and retry logic. Added logging for better traceability of processing and error handling.

Also included comprehensive unit tests for the new functionality, covering various scenarios such as empty prompts, maintaining order, handling rate limits, and testing with different adapters.

## Summary by CodeRabbit

* **New Features**
  * Added batch generation for processing multiple prompts with controlled concurrency, ordered results, per-prompt error handling, and automatic retries with exponential backoff and jitter.
* **Tests**
  * Added comprehensive test coverage validating batching, order preservation, concurrency limits, retry/backoff behavior, error handling, and integration with multiple adapters.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: benchwise/models.py [hunk 1]
```diff
@@ -24,7 +27,125 @@ def get_cost_estimate(self, input_tokens: int, output_tokens: int) -> float:
...
+            **kwargs: Additional arguments passed to generate()
+            
+        Returns:
+            List of responses in the same order as input prompts
+            
+        Example:
+            >>> adapter = OpenAIAdapter("gpt-3.5-turbo")
+            >>> prompts = ["Hello", "How are you?", "Tell me a joke"]
+            >>> responses = await adapter.batch_generate(prompts, max_concurrent=5)
+        """
```
[the remark this answers is not in the document]
Comment (yanurag-dev) [reply benchwise/models.py:60]: @coderabbit got it. will fix it

File: benchwise/models.py [hunk 2]
```diff
@@ -24,7 +27,125 @@ def get_cost_estimate(self, input_tokens: int, output_tokens: int) -> float:
...
+        # Gather all results with exception handling
+        results_with_indices = await asyncio.gather(*all_tasks, return_exceptions=True)
+        
+        # Handle any exceptions that weren't caught
+        processed_results = []
+        for item in results_with_indices:
+            if isinstance(item, Exception):
+                processed_results.append((len(processed_results), f"Error: {str(item)}"))
+            else:
+                processed_results.append(item)
```
[the remark this answers is not in the document]
Comment (yanurag-dev) [reply benchwise/models.py:146]: @coderabbit If exception occurred at index 2 so there would be two item already in processed_results and length will be 2, so exception will be added on index 2 not on 0. 

I'm quoting you on this 

Line 136 uses len(processed_results) as the index for uncaught exceptions, which assumes exceptions are processed in order and no previous exceptions occurred. However, if an exception occurs at index 2 and this is the first item processed, it would be assigned index 0 instead of 2, breaking order preservation.
